Add test vector for greasing in draft-ietf-privacypass-auth-scheme (#502)
ietf-wg-privacypass/base-drafts: draft-ietf-privacypass-auth-scheme.md

used to fit the document formatting constraints and not supported
in actual requests.

The last challenge on this list includes Basic authentication, a
greasing challenge, and a valid VOPRF challenge. Correct client
implementations will ignore both Basic and greasing challenges,

> while handling the last VOPRF challenge.

Author: the text has been updated accordingly.

used to fit the document formatting constraints and not supported
in actual requests.

> The last challenge on this list include Basic authentication, as well as a
> greasing challenge. Both MUST be ignored by client implementation.

Reviewer: ```suggestion
The last challenge on this list includes Basic authentication, as well as a
greasing challenge. Correct client implementations will ignore both challenges.
```

Author: I'm updating this paragraph to
```
The last challenge on this list includes Basic authentication, a greasing challenge, and a valid VOPRF challenge. Correct client implementations will ignore both Basic and greasing challenges, while handling the last VOPRF challenge.
```

LODAA5vcmlnaW4uZXhhbXBsZQ==", token-key="67H-0zgxA2HAjQx1dpaWcSluBem
aF9eSbfwopT-r1In6wPgryoYkmmaPOlv6s3TJ",unknownChallengeAttribute="ig
nore-me", max-age="10"

token-type-0: 0x0000
token-key-0: 856de3c710b892e7cca1ae5eb121af42ca8e779137a11224228c9b9
9b0729bf84d5057d030000309b8f0d06ccffa17561f9eacd4c312e985a6bc60ffbea
0610264dcb1726255313da81d665692686a1d8644f1516bf612cea009e6dff6d9a9a
959fb538e1b5b2343c092992942382bdde22d5b324b1e4618ed21d7806286c2ce
token-challenge-0: 0000acc3b25795c636fd9dd8b12982394abba8777d35978e8
77fc8848892a217233045ac25a3d55c07c54efe6372973fee0073e77fc61bf19ab88
0f20edf5d627502

> token-type-1: 0x0001

Reviewer: Why would a client ignore this challenge?

Author: client would ignore Basic and 0x0000. They should not ignore 0x0001.

Reviewer: Right, that's what I thought, but the text suggests that clients ought to ignore both.

Author: I seems github has not correctly assigned my comment https://github.com/ietf-wg-privacypass/base-drafts/pull/502#discussion_r1365766838. The text has been updated.

token-key-1: ebb1fed338310361c08d0c7576969671296e05e99a17d7926dfc28a
53fabd489fac0f82bca86249a668f3a5bfab374c9
max-age-1: 10
token-challenge-1: 0001000e6973737565722e6578616d706c65208a3e83a33d9
8005d2f30bef419fa6bf4cd5c6005e36b1285bbb4ccd40fa4b383000e6f726967696
e2e6578616d706c65

WWW-Authenticate: Basic realm="grease", PrivateToken challenge="AACs
w7JXlcY2_Z3YsSmCOUq7qHd9NZeOh3_IhIiSohcjMEWsJaPVXAfFTv5jcpc_7gBz53_G
G_GauIDyDt9dYnUC",token-key="hW3jxxC4kufMoa5esSGvQsqOd5E3oRIkIoybmbB
ym_hNUFfQMAADCbjw0GzP-hdWH56s1MMS6YWmvGD_vqBhAmTcsXJiVTE9qB1mVpJoah2
GRPFRa_YSzqAJ5t_22ampWftTjhtbI0PAkpkpQjgr3eItWzJLHkYY7SHXgGKGws4=",
PrivateToken challenge="AAEADmlzc3Vlci5leGFtcGxlIIo-g6M9mABdLzC-9Bn6
a_TNXGAF42sShbu0zNQPpLODAA5vcmlnaW4uZXhhbXBsZQ==", token-key="67H-0z
gxA2HAjQx1dpaWcSluBemaF9eSbfwopT-r1In6wPgryoYkmmaPOlv6s3TJ",unknownC

> hallengeAttribute="ignore-me", max-age="10"

Reviewer: How was this test vector generated?

Author: token-challenge-1 is the same as above.
token-challenge-0 with the below script on my local machine
```javascript
const hex_encode = (u) =>
  Array.from(u).map((b) => b.toString(16).padStart(2, 0)).join('')

const u8ToB64 = (u) =>
  btoa(String.fromCharCode(...u))

const b64ToB64URL = (s) =>
  s.replace(/\+/g, '-').replace(/\//g, '_')

const u8ToB64URL = (u) =>
  b64ToB64URL(u8ToB64(u))

const main = () => {
  const challenge = new Uint8Array(64+2)
  crypto.getRandomValues(challenge);
  challenge[0] = 0;
  challenge[1] = 0;
  console.log('token-challenge:', hex_encode(challenge))

  const key = new Uint8Array(128);
  crypto.getRandomValues(key);
  console.log('token-key:', hex_encode(key))

  const header = `PrivateToken challenge="${u8ToB64URL(challenge)}",token-key="${u8ToB64URL(key)}"`

  console.log('WWW-Authenticate:', header)
}

main()
```
